_rules/SVG-image-has-name

```diff
@@ -0,0 +1,103 @@
+---
+name: SVG image has name
+description: |
+  This rule checks that each SVG image element that is not marked as decorative, has an accessible name.
+
+success_criterion:
+- 1.1.1 # Non-Text Content
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Bryn Anderson
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any `SVG` `image` element, or any `SVG` element with the [semantic role](#semantic-role) of `img` that is [included in the accessibility tree](#included-in-the-accessibility-tree)
+
+### Expectation
+
+Each target element has a non-empty accessible name.
+
+## Assumptions
+
+*There are currently no assumptions*
+
+## Accessibility support
+
+ *There are no major accessibility support issues known for this rule.*
```

Comment:
Today, if the `svg` tag doesn't have the `role="img"` the **svg element is ignored by AT** (only tested with NVDA today, has to be confirmed in others AT), even if it has an accessible name correctly defined. Maybe that particularity has to be added in the accessibility support section ?

Reply:
The same result w/ VoiceOver. VoiceOver w/ the different situations:
```
<svg role="img"><title>Read</title><path></path></svg>
<svg role="img"><desc>Read</desc><path</path></svg>
<svg role="img"><title>Read</title><desc>Not Read</desc><path></path></svg>
<svg role="img"><desc>Not Read</desc><title>Read</title><path></path></svg>
<svg aria-labelledby="Title Desc" role="img"><title id="Title">1st Read</title><desc id="Desc">2nd Read</desc><path></path></svg>
<svg aria-labelledby="Title Desc" role="img"><desc id="Desc">2nd Read</desc><title id="Title">1st Read</title><path></path></svg>
<svg aria-labelledby="Title Desc" role="img"><title id="Title">1st Read</title><p id="Desc">2nd Read</p><path></path></svg>
<svg aria-labelledby="Title Desc"><title id="Title">Not Read</title><desc id="Desc">Not Read</desc><path></path></svg>
```

Reply:
This is good to know and I have tried to address this in a very general note under Accessibility Support without calling out specific browsers or AT.

Reply:
I refined the test and found new things: https://codepen.io/jorgefernandes/pen/LaYOmE 
´´´
<svg><title>título</title><desc>descrição</desc><path></path></svg> /* VO reads "título, group. descrição" */
<svg role="img"><title>título</title><desc>descrição</desc><path></path></svg> /* VO reads "título, image. título" */
´´´